### PR TITLE
Fix broken validation and multiple custom

### PR DESCRIPTION
**Pull request by davidstevens37**
created the ability pass an array of custom validations if needed.

Upgraded ember-cli, ember, and ember data so that the model validator will work on the newest release.

### CODE REVIEW

File: addon/mixins/model-validator.js [hunk 1]
```diff
@@ -37,7 +37,8 @@ export default Ember.Mixin.create({
       if(Object.keys(errors).length !== 0){
         var stateToTransition = this.get('isNew') ? 'created.uncommitted' : 'updated.uncommitted';
         this.transitionTo(stateToTransition);
-        store.recordWasInvalid(this, errors);
+				var recordModel = this.adapterDidInvalidate ? this : this._internalModel;
```
Comment (davidstevens37) [inline addon/mixins/model-validator.js]: This fixes the errors with 1.13.x and allows for current uses to still work. @esbanarango

File: addon/mixins/model-validator.js [hunk 2]
```diff
@@ -248,10 +252,10 @@ export default Ember.Mixin.create({
...
   },
   _getCustomValidator: function(validation){
-    var customValidator = validation.custom;
-    if (Ember.typeOf(validation.custom) === 'object' && validation.custom.hasOwnProperty('validation')) {
-      customValidator = validation.custom.validation;
-    } 
+    var customValidator = validation;
+    if (Ember.typeOf(validation) === 'object' && validation.hasOwnProperty('validation')) {
+      customValidator = validation.validation;
+    }
```
Comment (davidstevens37) [inline addon/mixins/model-validator.js]: this just changes how the function is passed a validation object and returns the custom validation funciton (to account for it being iterated over in the function calling it)

File: addon/mixins/model-validator.js [hunk 3]
```diff
@@ -47,14 +48,17 @@ export default Ember.Mixin.create({
...
+		for (var i = 0; i < validation.length; i++) {
+	    var customValidator = this._getCustomValidator(validation[i]);
+	    if (customValidator) {
+	      var passedCustomValidation = customValidator(property, this.get(property), this);
+	      if (!passedCustomValidation) {
+	        this.set('isValidNow', false);
+	        this._addToErrors(property, validation[i], Messages.customValidationMessage);
+	      }
+	    }
+		}
```
Comment (davidstevens37) [inline addon/mixins/model-validator.js]: this now ensures that validation is an array or converts it to one. then iterates over it and executes the custom validation as it did before, but for each in the array

**Comment by esbanarango** [comment]
@davidstevens37 Thank you very much for your work!! I review and merge this weekend.

**Comment by davidstevens37** [comment]
@esbanarango no problem, let me know if there is anything you think should be changed.

**Comment by davidstevens37** [comment]
@esbanarango is there anything in here you think should changed or is wrong?

**Pull request merged**